File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlFuncTableSource.cs
using System.Text;
using T1.Standard.IO;

namespace T1.SqlSharp.Expressions;

public class SqlFuncTableSource : SqlTableSource
{
    public new SqlType SqlType { get; } = SqlType.FuncTableSource;
    public required SqlFunctionExpression Function { get; set; }
    public override string ToSql()
    {
        var sql = new IndentStringBuilder();
        sql.Write(Function.ToSql());
        WriteSqlAfterTableName(sql);
        return sql.ToString();
    }
}

File: VimSharp/chrome-extension-sharp/src/t1-sqlts/Expressions/SqlFuncTableSource.ts
import { SqlType } from './SqlType';
import { TextSpan } from '../StringParser';
import { ISqlExpression, SqlVisitor } from './ISqlExpression';

export class SqlFuncTableSource implements ISqlExpression {
    SqlType: SqlType = SqlType.FuncTableSource;
    Span: TextSpan = new TextSpan();

    Accept(visitor: SqlVisitor): void {
        visitor.Visit_FunctionExpression(this);
    }

    ToSql(): string {
        return '';
    }
} 